src/NHibernate/Async/Criterion/QueryOver.cs

```diff
@@ -0,0 +1,69 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by AsyncGenerator.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+
+
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+
+using NHibernate.Criterion.Lambda;
+using NHibernate.Engine;
+using NHibernate.Impl;
+using NHibernate.SqlCommand;
+using NHibernate.Transform;
+
+namespace NHibernate.Criterion
+{
+	using System.Threading.Tasks;
+	public abstract partial class QueryOver<TRoot> : QueryOver, IQueryOver<TRoot>
+	{
+
+		private async Task<IList<TRoot>> ListAsync()
+		{
+			return await (criteria.ListAsync<TRoot>()).ConfigureAwait(false);
```

Comment:
You are correct, it seems that the "algorithm" to check if the returned type from the async method matches the method return type fails here. Will check and fix after I get home.